pkg/kube/services/services.go

```diff
@@ -92,6 +96,49 @@ func GetServiceURLFromMap(services map[string]*v1.Service, name string) string {
 	return GetServiceURL(services[name])
 }
 
+func getIstioVirtualService(namespace, name string) (*unstructured.Unstructured, error) {
+	f := kubeclient.NewFactory()
+	cfg, err := f.CreateKubeConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get kubernetes config")
+	}
+	dynamicClient, err := dynamic.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "error building dynamic clientset")
+	}
+
+	//  Create a GVR which represents an Istio Virtual Service.
+	virtualServiceGVR := schema.GroupVersionResource{
+		Group:    "networking.istio.io",
+		Version:  "v1alpha3",
+		Resource: "virtualservices",
+	}
+	virtualService, err := dynamicClient.Resource(virtualServiceGVR).Namespace(namespace).Get(context.TODO(), name, meta_v1.GetOptions{})
+	return virtualService, err
+}
+
+// GetUrlFromVirtualService finds url from virtual services istio
+func GetUrlFromVirtualService(virtualService *unstructured.Unstructured) (string, error) {
+	vs := virtualService.Object
+	if spec, ok := vs["spec"].(map[string]interface{}); ok {
+		if hosts, ok := spec["hosts"].([]interface{}); ok {
+			return fmt.Sprintf("%v", hosts[0]), nil
+		}
+	}
+	return "", errors.New("No url found in the virtual service")
+}
+
+func FindUrlFromVsIstio(namespace, name string) (string, error) {
```

Comment:
So, this is the only function we need to test as it uses the 2 other functions in the PR.